components/plugin_marketplace/marketplace_list/marketplace_list.tsx

```diff
@@ -55,9 +65,8 @@ export default class MarketplaceList extends React.PureComponent {
                         key={p.manifest.id}
```

Comment:
The line above causes the following error:
```
{ key: string; id: string; name: string; description: string; version: string; homepageUrl: string | undefined; releaseNotesUrl: string | undefined; labels: MarketplaceLabel[] | undefined; iconData: string | undefined; installedVersion: string | undefined; }' is missing the following properties from type 'MarketplaceItemProps': installing, isDefaultMarketplace, trackEvent, actions
```
Does someone have any idea was is causing this? I was expecting the `connect` call in `components/plugin_marketplace/marketplace_item/index.ts` to take care of assigning these props.

Reply:
@larkox Do you have any ideas what causes that?

Reply:
I see the issue. The `index.ts` file specifies that it should receive `MarketplaceItemProps` as its incoming props, and this type includes the props that are being injected by this function.

https://github.com/mattermost/mattermost-webapp/blob/e5f6e7ecd2c9a8f5e2fbf0cc8143ba72c87fb52a/components/plugin_marketplace/marketplace_item/index.ts#L18-L25

Changing it to this makes the type error go away
```ts
function mapStateToProps(state: GlobalState, props: {id: string}) {
```

but then the type checking doesn't catch errors when the parent is missing props being passed in. Not sure what needs to be done to properly type this function. @hmhealey Do you have any ideas?

Reply:
Thanks for catching that @mickmister. Your proposed solutions looks good. If the parent is missing a prop, TS will still throw an error. Hence, all cases are caught.

Fixed in 45fe7fd12.